File: StickyNavLayout/app/src/main/java/com/rg/stickynavlayout/MainActivity.java
package com.rg.stickynavlayout;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.LinearLayout;
import android.widget.ListView;
import android.widget.Toast;

import com.rg.stickynavlayout.view.StickyNavLayout;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    StickyNavLayout stickyNavLayout;
    ListView id_bottomview;
    MainAdapter adapter;
    LinearLayout id_topview;
    LinearLayout id_indicatorview;

    ArrayList<String> strings;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        strings=new ArrayList<>();

        stickyNavLayout= (StickyNavLayout) findViewById(R.id.stickyNavLayout);
        id_bottomview= (ListView) findViewById(R.id.id_bottomview);
        adapter=new MainAdapter(this, strings);
        id_bottomview.setAdapter(adapter);
        id_bottomview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Toast.makeText(MainActivity.this, "click3", Toast.LENGTH_SHORT).show();
            }
        });
        id_topview= (LinearLayout) findViewById(R.id.id_topview);
        id_topview.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "click", Toast.LENGTH_SHORT).show();
            }
        });
        id_indicatorview= (LinearLayout) findViewById(R.id.id_indicatorview);
        id_indicatorview.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "click2", Toast.LENGTH_SHORT).show();

                changeItems(13);
                stickyNavLayout.reset();
            }
        });

        changeItems(13);
    }

    private void changeItems(int num) {
        stickyNavLayout.setContentHeight(num);
        strings.clear();
        for (int i=0;i<num;i++) {
            strings.add("");
        }
        adapter.notifyDataSetChanged();
    }
}

File: SwipeRefreshLayoutDemo/app/src/main/java/com/renyu/swiperefreshlayoutdemo/MainActivity.kt
package com.renyu.swiperefreshlayoutdemo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.support.v4.widget.SwipeRefreshLayout
import com.renyu.swiperefreshlayoutdemo.adapter.MainAdapter
import com.renyu.swiperefreshlayoutdemo.view.BaseSwipeRefresh
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var adapter: MainAdapter? = null

    val beans: ArrayList<String> by lazy {
        ArrayList<String>()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        rv_main.setOnRefreshListener(SwipeRefreshLayout.OnRefreshListener {
            Handler().postDelayed({
                for (i in 0..19) {
                    beans.add(0, "s$i")
                }
                adapter?.notifyDataSetChanged()

                rv_main.setRefreshing(false, true)
            }, 3000)
        })
        rv_main.setOnLoadMoreListener(object : BaseSwipeRefresh.LoadMoreListener {
            override fun onRefresh() {
                Handler().postDelayed({
                    for (i in 0..19) {
                        beans.add("e$i")
                    }
                    adapter?.notifyDataSetChanged()

                    rv_main.setRefreshing(false, false)
                }, 3000)
            }
        })
        for (i in 0..19) {
            beans.add("$i")
        }
        adapter = MainAdapter(this, beans)
        rv_main.setAdapter(adapter!!)
    }
}
